src/main/java/com/aws/greengrass/smbridge/MessageBridge.java

```diff
@@ -51,26 +58,50 @@ private void handleMessage(Message message) {
         LOGGER.atDebug().kv("destinations", sourceDestinationMap).log("Message will be forwarded to destinations");
 
         if (sourceDestinationMap != null) {
-            List<TopicMapping.MappingEntry> destinations = sourceDestinationMap.get(sourceTopic);
-            if (destinations == null) {
-                return;
-            }
-            LOGGER.atDebug().log("HANDLING MESSAGE, NO IMPLEMENTED SM CLIENT");
-            /*
-            for (TopicMapping.MappingEntry destination : destinations) {
-                if (smClient != null) { // TODO: Add SM Client in Message Bridge
-                    Message msg = new Message(sourceTopic, message.getPayload());
+            final Consumer<TopicMapping.MappingEntry> processDestination = destination -> {
+                String stream = destination.getStream();
+                StreamMessage streamMessage;
+                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream(1024);
+
+                if (destination.isAppendTime()){
+                    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss.SSSSSS");
+                    LocalDateTime now = LocalDateTime.now();
+                    String stringTime = dtf.format(now) + ": ";
+                    byte[] time = stringTime.getBytes();
+                    try {
+                        byteArrayOutputStream.write(time);
```

Comment:
How is the customer supposed to parse this information? How do they know the encoded information? The length of the time string for example.

Reply:
The length of the time string could be documented, but the topic is variable-length. Perhaps a standard delimiter ("\r\n" or the like) could work?

Reply:
Please check newest commit